R/archivePackages.R

```diff
@@ -1,67 +1,93 @@
 
-##' The function moves older versions of packages into a CRAN-style 
+##' The function moves older versions of packages into a CRAN-style
 ##' archive folder.
-##' 
+##'
 ##' This function is still undergoing development and polish and may
 ##' change in subsequent versions.
 ##'
 ##' @title Move older copies of packages to an archive
+##' @importFrom stringr str_extract
 ##' @param repopath Character variable with the path to the repo;
 ##' defaults to the value of the \dQuote{dratRepo} option with
 ##' \dQuote{"~/git/drat"} as fallback
 ##' @param type Character variable for the type of repository, so far
-##'  \dQuote{source}, \dQuote{win.binary}, \dQuote{mac.binary}, 
+##'  \dQuote{source}, \dQuote{win.binary}, \dQuote{mac.binary},
 ##'  \dQuote{mac.binary.mavericks} or \dQuote{mac.binary.el-capitan}
 ##' @param pkg Optional character variable specifying a package name(s), whose
-##' older versions should be archived. If missing (the default), archiving is 
+##' older versions should be archived. If missing (the default), archiving is
 ##' performed on all packages.
-##' @param version R version information in the format \code{X.Y} or 
+##' @param version R version information in the format \code{X.Y} or
 ##'   \code{X.Y.Z}. Only used, if archiving binary packages.
-##'   (default: \code{version = getRversion()})
+##'   By default all R versions found in the respective CRAN-like directory
+##'   for the chosen `type` are checked.
 ##' @examples
 ##' \dontrun{
 ##'   archivePackages()   # archive all older package versions
 ##'   archivePackages(pkg = "drat")  # archive older copies of just one package
 ##' }
 ##' @author Thomas J. Leeper
 archivePackages <- function(repopath = getOption("dratRepo", "~/git/drat"),
-                            type = "source", 
-                            pkg,
-                            version = getRversion()) {
+                            type = c(
+                                "source", "mac.binary", "mac.binary.el-capitan",
+                                "mac.binary.mavericks", "win.binary"
+                            ),
+                            pkg = NULL,
+                            version = NULL) {
     
-    ## knows how to handle binary repos
-    repodir <- contrib.url2(repopath, type, version)
-
-    archive <- file.path(repodir, "Archive")
-    if (!file.exists(archive)) {
-        if (!dir.create(archive, recursive = TRUE)) {
-            stop("Archive directory not found and couldn't be created\n", call. = FALSE)
+    for (type in type) {
+        
+        # if no R version is supplied, the one from the
+        # respective "type" directory is scraped
+        if (is.null(version)) {
+            if (grepl("binary", type)) {
+                r_versions <- stringr::str_extract(list.dirs(gsub(
```

Comment:
I appreciate your offer to rewrite this and not use extra dependencies for a function that is a bit more tangential to the package.   

And if and when you rewite please avoid excess line breaks.   100 cols are fine.    You can see in the rest of the file that we often use close to 100 cols (and the one at 111 cols could be considered too long ...  but it works so no need for a PR or change of working code).